lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -84,24 +84,30 @@ def add_node_data(node)
   end
 
   # Rewrite a given file resource with the metadata from a fileserver based file
-  def replace_metadata(resource, metadata, recurse = false)
+  def replace_metadata(resource, metadata)
     if resource[:links] == "manage" && metadata.ftype == "link"
       resource.delete(:source)
-      resource[:ensure] = "link"
       resource[:target] = metadata.destination
-    elsif resource[:links] == "follow"
-      resource[:ensure] = metadata.ftype
     end
 
+    resource[:ensure] = metadata.ftype
```

Comment:
We definitely don't get the correct behavior; if the resource specifies a directory but the source is a file, the static catalog's behavior without this PR would cause a directory to be created.